google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/types/internal/JsonJavaTypeDescriptor.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019-2020 Google LLC
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+package com.google.cloud.spanner.hibernate.types.internal;
+
+import com.google.gson.Gson;
+import java.util.Properties;
+import org.hibernate.type.descriptor.WrapperOptions;
+import org.hibernate.type.descriptor.java.AbstractTypeDescriptor;
+import org.hibernate.usertype.DynamicParameterizedType;
+
+/**
+ * A {@link AbstractTypeDescriptor} for Spanner JSON columns.
+ */
+public class JsonJavaTypeDescriptor extends AbstractTypeDescriptor<Object>
+    implements DynamicParameterizedType {
+
+  private static final Gson gson = new Gson();
```

Comment:
What's the reason for using Gson? Hibernate uses Jackson. (See: https://github.com/vladmihalcea/hibernate-types/blob/master/hibernate-types-55/src/main/java/com/vladmihalcea/hibernate/type/util/ObjectMapperWrapper.java)

Reply:
We looked at this with @dzou and @zhumin8 -- gson is brought in through the client library, so it's already present.